### ISSUE TITLE
Initial Propagation of Timeline Impact Changes to Dependent Work Packages

### ISSUE THREAD

**Issue by jamescd18** [Northeastern-Electric-Racing/FinishLine#354]
## Overview

Whenever a timeline impact is implemented, propagate timeline shift to all dependent work packages to avoid users needing to implement numerous manual changes when timelines shift.

## Stakeholders

Product Stakeholder: @jamescd18 
Software Stakeholder: @RChandler234 
Reference Users: @zperry02 

## User Story

As a project lead, I want all subsequent work package timelines to adjust appropriately when I implement a timeline impact to the currently active work package without needing to implement additional changes.

## Success Metrics

Success of this feature will be measured by the time it takes a user to fully implement a change request with timeline impact changes on a project with around 5 subsequent work packages. 

Baseline: 3 minutes

## Rollout Plan

- Deploy to production
- Educate users and push them to add dependencies to work packages
- Measure to determine success level

## Out of Scope

tba

## Background / Context

Imagine a project with multiple work packages wherein each work package must be completed before the next one can start. Each work package would then have the previous work package listed in its dependencies list. The first work package is active and all subsequent work packages were set with a start date equal to their previous work package's end date. 

If you were to delay the active work package by adding duration or pushing its start date later, currently within FinishLine, none of the subsequent work packages would change in any way. This would lead to an inaccurate representation of the project's overall end date since the next work package after the active one could no longer be started at its set start date.

## Acceptance Criteria & Mock-ups

When a standard scope change request for work package A is accepted with a proposed solution with a timeline impact greater than zero, find all work packages in the same project that list work package A in their dependencies list and add the timeline impact # of weeks to their start dates. Also record these as implemented changes.

This adjustment to start dates needs to be performed recursively.
